velox/functions/prestosql/aggregates/tests/ChecksumAggregateTest.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <boost/algorithm/string/join.hpp>
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/prestosql/aggregates/tests/AggregationTestBase.h"
+
+using facebook::velox::exec::test::PlanBuilder;
+
+namespace facebook::velox::aggregate::test {
+
+class ChecksumAggregateTest : public AggregationTestBase {
+ protected:
+  static constexpr auto O = makeOptional<int64_t>;
+
+  template <typename T>
+  void assertSingleGroupChecksum(
+      const std::vector<std::optional<T>>& data,
+      const std::string& checksum) {
+    auto inputVector = makeNullableFlatVector<T>(data);
+    assertChecksum(inputVector, checksum);
+  }
+
+  void assertChecksum(
+      VectorPtr inputVector,
+      const std::string& expectedChecksum) {
+    auto rowVectors = std::vector{makeRowVector({inputVector})};
+
+    // DuckDB doesnt have checksum aggregation, so we will just pass in
+    // expected values to compare.
+    {
+      auto agg = PlanBuilder()
+                     .values(rowVectors)
+                     .partialAggregation({}, {"checksum(c0)"})
+                     .finalAggregation()
+                     .project({"to_base64(a0) as c0"})
+                     .planNode();
+      assertQuery(
+          agg,
+          fmt::format("VALUES (CAST(\'{}\' AS VARBINARY))", expectedChecksum));
```

Comment:
to_base64 returns a varchar; why do we cast expected checksum to varbinary here?